ctpdev : inputs shifts

lietava opened the pull request:
L0/L1 inputs shifted to LM latency

lietava committed:
dev: ctp inputs L0/L1 shifted to LM latency

lietava committed:
clang

lietava commented:
@shahor02 : please, have a look

shahor02 reviewed (dismissed):
@lietava I did not go into details of bitmask manipulations, but overall looks fine.
Is is ready to be merged?

lietava commented:
@shahor02 wait , please, for green light from me.

lietava committed:
dev: inputs decoding as option and new offset

lietava committed:
clang

lietava commented:
@shahor02 : 
0. I added new timeoffset   int64_t globalInputsShift  for inputs , the customOffset[16] is to be used for classes as before
1. added inputs decoding as option do CTP RawDataDecoder
2. I will add same decoding to CTF decoder as we have discussed.
This gives us both options. ( I am on holiday next week starting from tomorrow so I will modify CTF decoder after).
Please, merge this PR if ok.

shahor02 commented on Detectors/CTP/reconstruction/src/RawDataDecoder.cxx:
```diff
@@ -286,15 +290,93 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
   if (mStickyError) {
     if (nwrites < mErrorMax) {
-      std::string file = "/tmp/dumpCTP" + std::to_string(nwrites) + ".bin";
+      std::string file = "dumpCTP" + std::to_string(nwrites) + ".bin";
```
Why not making the output directory settable? See e.g. how it was done for the ITS:

https://github.com/AliceO2Group/AliceO2/blob/e9e04179907dfea3486a21092ee8f44664a51800/Detectors/ITSMFT/common/workflow/src/STFDecoderSpec.cxx#L391

https://github.com/AliceO2Group/AliceO2/blob/e9e04179907dfea3486a21092ee8f44664a51800/Detectors/ITSMFT/common/workflow/src/STFDecoderSpec.cxx#L91-L96

and https://github.com/AliceO2Group/AliceO2/blob/e9e04179907dfea3486a21092ee8f44664a51800/Detectors/ITSMFT/common/reconstruction/src/RawPixelDecoder.cxx#L521-L522

shahor02 reviewed (approved):
Hi @lietava 

Looks fine, but I would make the dump directory configurable (can be done in a separate commit), let me know if you want me to merge this as is.

lietava commented:
HI @shahor02 , actually the dump was already merged yesterday  in separate PR(11716) - I wanted to have it in P2 next Monday.  I will add configurable dir later in separate commit - thanks for examples.
Please, merge this as it is.

shahor02 commented:
ok, will merge once CI is passed.

lietava commented:
@shahor02 : any errors related to ctp ?
